openerp/addons/base/ir/ir_attachment.py

```diff
@@ -427,8 +430,26 @@ def create(self, cr, uid, values, context=None):
             values.pop(field, False)
         values = self._check_contents(cr, uid, values, context=context)
         self.check(cr, uid, [], mode='write', context=context, values=values)
+        if 'url' in values:
+            self._get_asset_attachments.clear_cache(self)
         return super(ir_attachment, self).create(cr, uid, values, context)
 
     def action_get(self, cr, uid, context=None):
         return self.pool.get('ir.actions.act_window').for_xml_id(
             cr, uid, 'base', 'action_attachment', context=context)
+
+    @api.model
+    @tools.ormcache('version', 'xmlid', 'inc', 'asset_type')
+    def _get_asset_attachments(self, version, xmlid, inc, asset_type):
+        print version, xmlid, inc, asset_type
+        """return assets for a view given by its xmlid"""
+        domain = [
+            # this first proposition is not really necessary, but nudges
+            # postgres into using the index on url
+            ('url', '!=', False),
```

Comment:
without this, postgres didn't use the index in some cases, with it, always. Do you have different results? If postgres 9.4 here